src/bootstrap/bootstrap.py

```diff
@@ -925,28 +925,20 @@ def update_submodules(self):
         ).decode(default_encoding).splitlines()]
         filtered_submodules = []
         submodules_names = []
-        llvm_checked_out = os.path.exists(os.path.join(self.rust_root, "src/llvm-project/.git"))
-        external_llvm_provided = self.get_toml('llvm-config') or self.downloading_llvm()
-        llvm_needed = not self.get_toml('codegen-backends', 'rust') \
-            or "llvm" in self.get_toml('codegen-backends', 'rust')
         for module in submodules:
+            # This is handled by native::Llvm in rustbuild, not here
             if module.endswith("llvm-project"):
-                # Don't sync the llvm-project submodule if an external LLVM was
-                # provided, if we are downloading LLVM or if the LLVM backend is
-                # not being built. Also, if the submodule has been initialized
-                # already, sync it anyways so that it doesn't mess up contributor
-                # pull requests.
```

Comment:
This sync is no longer done with this PR I believe when using `./x.py check` instead of `./x.py build`.

Reply:
That seems fine though? `x.py check` shouldn't require building LLVM at all.

Reply:
But if the user pulls master with a LLVM update, makes some changes, does `./x.py check` and then `git add -a`, the commit will contain a submodule change as the LLVM submodule wasn't updated. This is what is meant with "messing up contributor pull requests".

Reply:
Hmm well I'm not 100% convinced how useful that is, but I'll change it for now to be consistent with bootstrap.py

Reply:
Actually this doesn't work with `init-repo.sh`. Even if I call `update_llvm_submodule` unconditionally, if `src/llvm-project/.git` exists, it will skip the check unconditionally: https://github.com/rust-lang/rust/blob/6bda460cb0d21acef3cf44cc2ea8b379abc7d5c4/src/bootstrap/native.rs#L104-L109

I don't know how to fix this :/

Reply:
Oh! I'm being dumb, `init-repo.sh` sets `submodules = false`, I can just use that.

Reply:
It's critical that we update the submodule if it exists, independent of whether the LLVM step is hit.